### PR TITLE
Add integrate pyflakes enhanced AST into coala project

### PR DESCRIPTION
**Pull request by srisankethu**
@jayvdb Can you check this? Like you suggested I did not add the dead code fixing part.

### CODE REVIEW

File: data/projects.js [hunk 1]
```diff
@@ -401,7 +401,7 @@ coala should support generating metrics for your code.",
 		"markdown": "generic_bears.md",
 		"collaborating_projects" : ["coala"]
 	},
-	{
+        {
```
Comment (sils) [inline data/projects.js]: why this change?

File: data/projects.js [hunk 2]
```diff
@@ -419,5 +419,23 @@ coala should support generating metrics for your code.",
 		"tags" : ["Jetbrains IDEs", "Plugins"],
 		"markdown": "jetbrains-ide-plugin.md",
 		"collaborating_projects" : ["coala", "jetbrains"]
+	},
+        {
```
Comment (sils) [inline data/projects.js]: the rest of the file uses tabs it seems

**Comment by sils** [comment]
not fully convinced yet, project idea doesn't seem competely clearly lined out and lacks a few more details. Also sounds rather like a low project to me, just writing three wrappers for things?

File: data/projects/integrate-pyflakes-AST.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#### Milestones
+
+##### GSOC 2017 CODING STARTS
+
+* The participate knows how to integrate pyflakes-AST into coala
+* The participant knows how to write coala-bears
+
+##### GSOC 2017 MIDTERM
+
+* Integrate pyflakes-enhanced AST into coala
```
Comment (jayvdb) [inline data/projects/integrate-pyflakes-AST.md]: I think this pyflakes enhanced AST should be a "meta-bear", like `AnnotationBear` and `CLangBear`, i.e. `PyflakesASTBear`, and the wrapped flake8 extensions should then depend on `PyflakesASTBear`.

File: data/projects/integrate-pyflakes-AST.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Integrate the [`pyflakes`](https://pypi.python.org/pypi/pyflakes)-enhanced [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) into coala so that bears can access it, and create two bears that wrap the flake8 AST plugins [`flake8-future-import`](https://github.com/xZise/flake8-future-import) and [`flake8-string-format`](https://github.com/xZise/flake8-string-format) using the pyflakes-enhanced AST.
```
Comment (jayvdb) [inline data/projects/integrate-pyflakes-AST.md]: We can add that these flake8 plugins need to be installed into a dynamically created venv.
https://gitlab.com/coala/package_manager/issues/25

File: data/projects.js [hunk 5]
```diff
@@ -419,5 +419,24 @@ coala should support generating metrics for your code.",
...
+                                a meta-bear i.e PyflakesASTBear",
+				"Create two bears that wrap the flake8 AST plugins \
+				'flake8-future-import' and 'flake8-string-format' \
+                                using the PyflakesASTBear."
+		],
+		"difficulty" : "medium",
+		"issues" : [],
+		"mentors" : ["jayvdb", "srisankethu"],
+		"initiatives" : ["GSoC", "RGSoC"],
+		"tags" : ["Bears"],
```
Comment (sims1253) [inline data/projects.js]: maybe add pyflakes?

File: data/projects/integrate-pyflakes-AST.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+Integrate the [`pyflakes`](https://pypi.python.org/pypi/pyflakes)-enhanced [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) into coala so that bears can access it by creating a meta bear like [AnnotationBear](https://github.com/coala/coala-bears/blob/1892b64722f1c10bd2a29611af4dcc18ad76af3a/bears/general/AnnotationBear.py) and [ClangBear](https://github.com/coala/coala-bears/blob/27a06d19076633a4e796472cf6d6ad55c27045c5/bears/c_languages/ClangBear.py), i.e PyflakesASTBear. Create two bears that wrap the flake8 AST plugins [`flake8-future-import`](https://github.com/xZise/flake8-future-import) and [`flake8-string-format`](https://github.com/xZise/flake8-string-format) using the PyflakesASTBear. The flake8 plugins needed to installed can be added to a dynamically created venv. Check out [this issue](https://gitlab.com/coala/package_manager/issues/25).
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: This needs linebreaks and also some structure. Give it an introduction part and a "this is the idea what to do" part

Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: and a "this is the goal" part

File: data/projects/integrate-pyflakes-AST.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+#### Milestones
+
+##### GSOC 2017 CODING STARTS
+
+* The participant knows how to write coala-bears and has written coala-bears before
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: maybe have a small trivial bear as requirement. also look at the milestones for other projects.

File: data/projects/integrate-pyflakes-AST.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+Integrate the [`pyflakes`](https://pypi.python.org/pypi/pyflakes)-enhanced [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) into coala so that bears can access it by creating a meta bear like [AnnotationBear](https://github.com/coala/coala-bears/blob/1892b64722f1c10bd2a29611af4dcc18ad76af3a/bears/general/AnnotationBear.py) and [ClangBear](https://github.com/coala/coala-bears/blob/27a06d19076633a4e796472cf6d6ad55c27045c5/bears/c_languages/ClangBear.py), i.e PyflakesASTBear. Create two bears that wrap the flake8 AST plugins [`flake8-future-import`](https://github.com/xZise/flake8-future-import) and [`flake8-string-format`](https://github.com/xZise/flake8-string-format) using the PyflakesASTBear. The flake8 plugins needed to installed can be added to a dynamically created venv. Check out [this issue](https://gitlab.com/coala/package_manager/issues/25).
+
+#### Milestones
+
+##### GSOC 2017 CODING STARTS
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: Community Bonding

File: data/projects/integrate-pyflakes-AST.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+* The participant knows how to write coala-bears and has written coala-bears before
+
+##### GSOC 2017 MIDTERM
+
+* Integrate pyflakes-enhanced AST into coala by writing a meta bear i.e PyflakesASTBear
+
+##### GSOC 2017 FINAL
+
+* Create two bears that wrap flake8 AST plugins mentioned using PyflakesASTBear
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: The milestones need a lot of work. They have to be very good. Look at other projects for help

File: data/projects.js [hunk 10]
```diff
@@ -419,5 +419,24 @@ coala should support generating metrics for your code.",
...
+		"markdown": "integrate-pyflakes-AST.md",
+		"collaborating_projects" : ["coala"]
```
Comment (jayvdb) [inline data/projects.js]: we can add 'pyflakes' here.  I am a member of their team.

File: data/projects/integrate-pyflakes-AST.md (hunk 7 above)
Comment (jayvdb) [inline data/projects/integrate-pyflakes-AST.md]: Solve an 'easy' pyflakes bug also.
https://bugs.launchpad.net/pyflakes/+bugs?field.tag=easy

File: data/projects.js [hunk 11]
```diff
@@ -419,5 +419,24 @@ coala should support generating metrics for your code.",
 		"tags" : ["Jetbrains IDEs", "Plugins"],
 		"markdown": "jetbrains-ide-plugin.md",
 		"collaborating_projects" : ["coala", "jetbrains"]
+	},
+	{
+		"name" : "Integrate pyflakes-enhanced AST into coala ",
+		"desc" : "Integrate pyflakes-enhanced AST into coala \
+		so that bears can access it.",
+		"requirements" : [
```
Comment (sims1253) [inline data/projects.js]: These are requirements for getting the project, not the goals of the project

File: data/projects/integrate-pyflakes-AST.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+#### Milestones
+
+##### GSOC 2017 COMMUNITY BONDING
+
+* The participant has done prior contributions by writing and testing coala-bears. Check out [existing bear proposal](https://github.com/coala/coala-bears/issues?q=is%3Aissue+is%3Aopen+label%3A%22bear+proposal%22).
+* The participant has done prior contributions to `Pyflakes` by solving alteast one [easy task](https://bugs.launchpad.net/pyflakes/+bugs?field.tag=easy).
+
+##### GSOC 2017 MIDTERM
+
+* Integrate pyflakes-enhanced AST into coala by writing a meta bear i.e PyflakesASTBear and test the integration.
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: Has to be completly tested and documented. Also maybe add mroe milestones to offer more points of reference

File: data/projects/integrate-pyflakes-AST.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+##### GSOC 2017 FINAL
+
+* Write and test a bear that wraps `flake8-future-import` using PyflakesASTBear.
+* Write and test a bear that wraps `flake8-string-format` using PyflakesASTBear.
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: also missing documentation.
Also aren't wrapping bears rather easy so shouldnt this be the second step and the third step would be to add the ast and make a bear using it?

File: data/projects/integrate-pyflakes-AST.md [hunk 14]
```diff
@@ -0,0 +1,21 @@
+This project focuses on integrating the [`pyflakes`](https://pypi.python.org/pypi/pyflakes)-enhanced [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) into coala and wrap flake8 AST plugins [`flake8-future-import`](https://github.com/xZise/flake8-future-import) and [`flake8-string-format`](https://github.com/xZise/flake8-string-format) using it.
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: This needs a little story as description. What problem are we trying to solve with this project? How does it help coala and the users of coala? 
We need a motivation for the project (as in why do we want this) and a goal for this.

Comment (jayvdb) [inline data/projects/integrate-pyflakes-AST.md]: The problem we are trying to solve:

pycodestyle supports plugins running AST based checks u.
flake8 has many plugins which are commonly used in the python development community.  flake8 is a wrapper, similar to coala, that provides that use the python-enhanced AST to its plugins.  Instead of coala using the flake8 wrapper, which adds another level of complexity and uncertainty with regards to configuration, coala would like to directly load the python-enhanced AST and invoke the flake8 plugins.
This allows the coala and flake8 communities to cooperate on plugin development using common plugin framework.

Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: So the problem is: Complexity due to additional abstraction layer and division of communities due to it

File: data/projects.js [hunk 15]
```diff
@@ -419,5 +419,27 @@ coala should support generating metrics for your code.",
...
+                                 "The participant has done prior contributions \
```
Comment (sims1253) [inline data/projects.js]: To simplify this use "The participant has a patch to any bear merged"

File: data/projects.js [hunk 16]
```diff
@@ -419,5 +419,27 @@ coala should support generating metrics for your code.",
...
+                                 by writing and testing coala-bears. Check out \
+                                 [existing bear proposal](https://github.com/coa\
+                                 la/coala-bears/issues?q=is%3Aissue+is%3Aopen+la\
+                                 bel%3A%22bear+proposal%22).",
+                                 "The participant has done prior contributions \
```
Comment (sims1253) [inline data/projects.js]: Similar to above, use "The participant has a patch to pyflakes merged"

Comment (sims1253) [inline data/projects.js]: keeping the links of course :)

File: data/projects/integrate-pyflakes-AST.md [hunk 17]
```diff
@@ -0,0 +1,24 @@
+flake8 has many plugins which are commonly used in the python development community.  flake8 is a wrapper, similar to coala, that provides that use the python-enhanced AST to its plugins.  Instead of coala using the flake8 wrapper, which adds another level of complexity and uncertainty with regards to configuration, coala would like to directly load the python-enhanced AST and invoke the flake8 plugins.
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: linebreaks please :D

Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: "... that provides the python-enhanced AST to its plugins." The "that use" is not needed.

File: data/projects/integrate-pyflakes-AST.md [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+This allows the coala and flake8 communities to cooperate on plugin development using common plugin framework.
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: "using a common plugin framework." missing a

File: data/projects/integrate-pyflakes-AST.md [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+
+This project focuses on integrating the [`pyflakes`](https://pypi.python.org/pypi/pyflakes)-enhanced [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) into coala and wrap flake8 AST plugins [`flake8-future-import`](https://github.com/xZise/flake8-future-import) and [`flake8-string-format`](https://github.com/xZise/flake8-string-format) using it.
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: linebreaks again :D

File: data/projects/integrate-pyflakes-AST.md [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+
+This can be done by creating a meta bear like [AnnotationBear](https://github.com/coala/coala-bears/blob/1892b64722f1c10bd2a29611af4dcc18ad76af3a/bears/general/AnnotationBear.py) and [ClangBear](https://github.com/coala/coala-bears/blob/27a06d19076633a4e796472cf6d6ad55c27045c5/bears/c_languages/ClangBear.py), say PyflakesASTBear. Then use this bear to wrap the flake8 AST plugins. Tutorials for writing and testing bears can be found in our [documentation](https://api.coala.io).
+
+The flake8 plugins needed to installed can be added to a dynamically created venv. Check out [this issue](https://gitlab.com/coala/package_manager/issues/25).
+
+#### Milestones
+
+##### GSOC 2017 COMMUNITY BONDING
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: Can you update the milestones to the new format similar to http://projects.coala.io/#/projects?project=integrate_antlr_v4_into_coala (4 phases)

**Comment by srisankethu** [comment]
@jayvdb Any more milestones that I need to add?

File: data/projects/integrate-pyflakes-AST.md [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+
+* The participant has done prior contributions by writing and testing coala-bears. Check out [existing bear proposal](https://github.com/coala/coala-bears/issues?q=is%3Aissue+is%3Aopen+label%3A%22bear+proposal%22).
```
Comment (sims1253) [inline data/projects/integrate-pyflakes-AST.md]: In this phase either a concept or a cEP (depending on the severity of changes to be done) should be written and finished. I'd say a concept is enough. So have something like "Api concept exists"

File: data/projects.js [hunk 22]
```diff
@@ -459,5 +459,25 @@ coala should support generating metrics for your code.",
...
 		"markdown": "documentation_extraction.md",
 		"collaborating_projects" : ["coala"]
+	},
+	{
+		"name" : "Integrate pyflakes-enhanced AST into coala ",
+		"desc" : "Integrate pyflakes-enhanced AST into coala \
+		so that bears can access it.",
+		"requirements" : [
+                                 "The participant has a patch to any bear merged. Check out \
+                                 [existing bear proposal](https://github.com/coa\
```
Comment (sims1253) [inline data/projects.js]: this is a list of new bear proposals. I think as a requirement it is enough to also just fix a bug.

**Comment by sims1253** [comment]
I think we can merge it after this round

**Comment by sims1253** [comment]
ack 15b6bd1531b3897e506740fc9b8fba68734b5ce5
